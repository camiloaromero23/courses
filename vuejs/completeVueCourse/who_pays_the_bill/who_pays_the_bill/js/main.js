new Vue({
	el: '#app',
	data() {
		return {
			state: true,
			inputName: '',
			names: [],
			showError: false,
			loserName: '',
		};
	},
	computed: {
		log() {
			console.log('Halooo');
		},
	},
	methods: {
		addName() {
			if (this.validateName(this.inputName)) {
				this.showError = false;
				this.names = [...this.names, this.inputName];
				this.inputName = '';
			} else {
				this.showError = true;
			}
		},
		validateName(value) {
			if (value !== '') {
				return true;
			}
			return false;
		},
		checkLoser() {
			this.loserName = this.names[
				Math.floor(Math.random() * this.names.length)
			];
			this.changeAppWindow();
		},
		changeAppWindow() {
			this.state = !this.state;
		},
		resetApp() {
			this.changeAppWindow();
			this.names = [];
		},
		removeName(index) {
			this.names.splice(index, 1);
		},
	},
});
