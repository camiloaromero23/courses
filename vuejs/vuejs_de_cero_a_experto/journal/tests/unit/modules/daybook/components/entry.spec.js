import { shallowMount } from '@vue/test-utils';

import Entry from '@/modules/daybook/components/Entry.vue';

import { journalState } from '../../../../mock-data/test-journal-state.js';

describe('Entry component tests', () => {
  const mockRouter = {
    push: jest.fn(),
  };
  const mockEntry = journalState.entries[0];
  let wrapper = shallowMount(Entry, {
    props: {
      entry: mockEntry,
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect at clicking inside the entry-container', () => {
    const entryContainer = wrapper.find('.entry-container');
    entryContainer.trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: mockEntry.id },
    });
  });

  it('should evaulate computed properties correctly', () => {
    const { yearDay, month, day } = wrapper.vm;
    console.log(yearDay);
    expect(day).toBe(15);
    expect(month).toBe('September');
    expect(yearDay).toBe('2021, Wednesday');
  });
});
