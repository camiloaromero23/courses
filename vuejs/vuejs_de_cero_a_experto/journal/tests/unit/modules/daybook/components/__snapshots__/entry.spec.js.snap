// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component tests should match the snapshot 1`] = `
<div class="mb-3 pointer p-2 entry-container">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">15</span><span class="mx-1 fs-5">September</span><span class="mx-2 fw-light">2021, Wednesday</span></div>
  <div class="entry-description">Learning Vue...</div>
</div>
`;
